### PR TITLE
Update to Letsencrypts new R3 intermediate

### PR DESCRIPTION
**Pull request by mhummels**
Updated the intermediate download to R3 one and replace the renewed cert only when new one is received successfully

### CODE REVIEW

File: renew_certificate.sh [hunk 1]
```diff
@@ -51,9 +51,11 @@ echo "Started python HTTP server with pid $pid"
 
 export SSL_CERT_FILE=cacert.pem
 "$PYTHON" acme-tiny/acme_tiny.py --account-key letsencrypt/account.key --csr letsencrypt/domain.csr --acme-dir tmp-webroot/.well-known/acme-challenge > letsencrypt/signed.crt.tmp
-mv letsencrypt/signed.crt.tmp letsencrypt/signed.crt
+if [ -f letsencrypt/signed.crt.tmp ] && [ -s letsencrypt/signed.crt.tmp ]; then
+    mv letsencrypt/signed.crt.tmp letsencrypt/signed.crt
+fi
 echo "Downloading intermediate certificate..."
-wget --no-verbose --secure-protocol=TLSv1_2 -O - https://letsencrypt.org/certs/lets-encrypt-x3-cross-signed.pem > letsencrypt/intermediate.pem
+wget --no-verbose --secure-protocol=TLSv1_2 -O - https://letsencrypt.org/certs/lets-encrypt-r3-cross-signed.pem > letsencrypt/intermediate.pem
```
Comment (Yannik) [inline renew_certificate.sh:58]: i will merge this

File: renew_certificate.sh [hunk 2]
```diff
@@ -51,9 +51,11 @@ echo "Started python HTTP server with pid $pid"
 
 export SSL_CERT_FILE=cacert.pem
 "$PYTHON" acme-tiny/acme_tiny.py --account-key letsencrypt/account.key --csr letsencrypt/domain.csr --acme-dir tmp-webroot/.well-known/acme-challenge > letsencrypt/signed.crt.tmp
-mv letsencrypt/signed.crt.tmp letsencrypt/signed.crt
+if [ -f letsencrypt/signed.crt.tmp ] && [ -s letsencrypt/signed.crt.tmp ]; then
+    mv letsencrypt/signed.crt.tmp letsencrypt/signed.crt
+fi
```
Comment (Yannik) [inline renew_certificate.sh:56]: please you undo this change

**Review by Yannik — CHANGES_REQUESTED** [review]

**Comment by mhummels** [comment]
Hi,

I can remove that but why?

I had some problems with letsencrypt Not issuing certificates. In that case
the resulting signed.crt.tmp is not there or empty. The only thing that if
does, is checking if the file exists and is not empty. So it is just makes
sure the renewed certificate is valid before replacing the probably still
working one.

I am very interested to hear why this change should be removed.

BRs,
Michael


Yannik Sembritzki <notifications@github.com> schrieb am Fr., 22. Jan. 2021,
18:52:

File: renew_certificate.sh (hunk 2 above)
Comment (mhummels) [reply renew_certificate.sh:56]: I can remove that but why?

I had some problems with letsencrypt Not issuing certificates. In that case
the resulting signed.crt.tmp is not there or empty. The only thing that if
does, is checking if the file exists and is not empty. So it is just makes
sure the renewed certificate is valid before replacing the probably still
working one.

I am very interested to hear why this change should be removed.

Comment (Yannik) [reply renew_certificate.sh:56]: we can talk about it in a separate issue. it's a change completely unrelated to the update of the intermediate cert.
